openpype/plugins/publish/submit_gather_deadline.py

```diff
@@ -0,0 +1,127 @@
+        self._comment = context.data.get("comment", "")
+        self._deadline_user = context.data.get( "deadlineUser", getpass.getuser())
+
+        # get output path
+        render_path = instance.data["publishDir"]
+        self.log.info(f">>>> Render path is {render_path}")
+        context.data["currentFile"] = ""
+
+        legacy_io.Session["AVALON_TASK"] = instance.data['task']
+        legacy_io.Session["AVALON_ASSET"] = instance.data['asset']
+
+
+        batch = f"{instance.data['project']} - {instance.data['asset']} - "
+        batch += f"{instance.data['task']} - {instance.data['subset']} - "
+        batch += f"v{instance.data['version']:03}"
+
+        instance.data["deadlineSubmissionJob"] = {
+            "Props" : {
+                "Batch": batch,
+                "User": context.data.get("deadlineUser", getpass.getuser()),
+                }
+        }
+        
+        instance.data["farm"] = True
+        instance.data["outputDir"] = os.path.dirname(
+            render_path).replace("\\", "/")
+        instance.data["publishJobState"] = "Suspended"
+        instance.context.data["version"] = instance.data["version"]
+
+        for k, v in instance.data["published_representations"].items():
+            self.expected_files(
+                instance,
+                v["published_files"][0],
+                instance.data["frameStart"],
+                instance.data["frameEnd"]
+            )
+
+        self.redefine_families(instance, families)
+    
+    def expected_files(
+        self,
+        instance,
+        path,
+        start_frame,
+        end_frame
+    ):
+        """ Create expected files in instance data """
+
+        if not instance.data.get("expectedFiles"):
+            instance.data["expectedFiles"] = []
+
+        dirname = os.path.dirname(path)
+        file = os.path.basename(path)
+
+        if "#" in file:
+            pparts = file.split("#")
+            padding = "%0{}d".format(len(pparts) - 1)
+            file = pparts[0] + padding + pparts[-1]
+
+        if "%" not in file:
+            instance.data["expectedFiles"].append(path)
+            return
+
+        if instance.data.get("slate"):
+            start_frame -= 1
+
+        for i in range(start_frame, (end_frame + 1)):
+            instance.data["expectedFiles"].append(
+                os.path.join(dirname, (file % i)).replace("\\", "/"))
+
+    @staticmethod
+    def redefine_families(instance, families):
+        """This method changes the family into 'write' for nuke so it is that
+        it is not recogniced by following plugins and thus they are turned off."""
+
+        if "render.farm" in families:
+            instance.data['family'] = 'write'
+            families.insert(0, "render2d")
+        elif "prerender.farm" in families:
+            instance.data['family'] = 'write'
+            families.insert(0, "prerender")
+        elif "gather.farm" in families:
+            instance.data['family'] = 'write'
```

Comment:
mmh but if this is going to be renamed as write how can we intercept the gather.farm and turn it into a normal gather in the family afterwards?

Reply:
For now I just left it like that to replicate what the OP team did with Nuke; I don't know whether we actually need to turn this back into gather, is there a reason why? the publish seems OK if I am not mistaken.
As there are many moving pieces here, after I am done with the parts I am currently testing I will revisit this then.

Reply:
well, the actual part of this is that i fear that the templated transcodes will not get picked up with the correct profiles. There is a nuke profile for render, but we should not operate under the assumption that it's the same as the gather.
Therefore, what you did in getting the family as write here works, but then it gets switched as "render", not gather, which can have a different profile.

Reply:
tested on your logs and it does actually get converted to "review" family. We need to convert to gather to make this work consistently